roles/etcd/templates/etcd.j2

```diff
@@ -35,7 +35,15 @@ start)
     {% endmacro -%}
 
     {% macro init_cluster() -%}
-    export ETCD_INITIAL_CLUSTER_STATE=new
+
+    if [ ! -f /var/tmp/etcd.existing ]
+    then
+      export ETCD_INITIAL_CLUSTER_STATE=new
+      touch /var/tmp/etcd.existing
+    else
+      export ETCD_INITIAL_CLUSTER_STATE=existing
+    fi
+
```

Comment:
2.1 to 2.2 is a seamless upgrade. 
https://github.com/coreos/etcd/releases/tag/v2.2.0

On 9 Dec 2015, at 21:57, jojimt wrote:

>> @@ -35,7 +35,15 @@ start)
>>   {% endmacro -%}
>>
>>   {% macro init_cluster() -%}
>> -    export ETCD_INITIAL_CLUSTER_STATE=new
>> +
>> +    if [ ! -f /var/tmp/etcd.existing ]
>> +    then
>> +      export ETCD_INITIAL_CLUSTER_STATE=new
>> +      touch /var/tmp/etcd.existing
>> +    else
>> +      export ETCD_INITIAL_CLUSTER_STATE=existing
>> +    fi
>> +
>
> Would this impact an etcd upgrade? Let's see we are going from version 
> 2.1 to 2.2. In that case, should we clean this up?
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/contiv/ansible/pull/4/files#r47191601